### PR TITLE
CSP-558 Redirect to return url if available

### PR DESCRIPTION
**Pull request by DevenKShah**

### CODE REVIEW

**Review by MarkFCain — APPROVED** [review]

File: src/SFA.DAS.ApprenticeAccounts.Web/Pages/Account.cshtml.cs [hunk 1]
```diff
@@ -39,9 +40,11 @@ public AccountModel(ApprenticeApi api, NavigationUrlHelper urlHelper)
         public bool IsCreating { get; private set; } = false;
 
         public async Task<IActionResult> OnGetAsync(
-            [FromServices] AuthenticatedUser user)
+            [FromServices] AuthenticatedUser user,
+            [FromQuery] string returnUrl)
         {
             ViewData.SetWelcomeText("Welcome");
+            TempData[ReturnUrlKey] = returnUrl;
```
Comment (Pauljgraham) [inline src/SFA.DAS.ApprenticeAccounts.Web/Pages/Account.cshtml.cs:47]: What happens when the returnUrl is blank or an empty string or null. Will it still be saved to the TempData?

Comment (DevenKShah) [reply src/SFA.DAS.ApprenticeAccounts.Web/Pages/Account.cshtml.cs:47]: Yes, I need the temp data set anyway so I can make decision on post.

Comment (Pauljgraham) [reply src/SFA.DAS.ApprenticeAccounts.Web/Pages/Account.cshtml.cs:47]: Well if the RedirectURL ever ends up being an empty string. we will redirect he user to a blank page (ie if TempData.Get returns true, we will end up redirecting).  


Looking at it a bit closer. The postback is on the same page, so we don't have to use TempData. The returnUrl is still in the URL when we do OnPost(), So we can just check it on PostBack and redirect accordingly. 

I believe the general consensus is not to use TempData if we don't need to

Comment (DevenKShah) [reply src/SFA.DAS.ApprenticeAccounts.Web/Pages/Account.cshtml.cs]: @Pauljgraham This is fixed.

File: src/SFA.DAS.ApprenticeAccounts.Web/Pages/Account.cshtml.cs [hunk 2]
```diff
@@ -13,6 +13,7 @@ namespace SFA.DAS.ApprenticeAccounts.Web.Pages
     [HideNavigationBar]
     public class AccountModel : PageModel
     {
+        const string ReturnUrlKey = "returnUrl";
```
Comment (Pauljgraham) [inline src/SFA.DAS.ApprenticeAccounts.Web/Pages/Account.cshtml.cs:16]: I don't think that var will be needed now (it's just noise)

**Review by Pauljgraham — APPROVED** [review]
Approved, with just a tiny suggestion about code clean up

**Pull request merged**